internal/testutil/assert.go

```diff
@@ -0,0 +1,165 @@
+package testutil
+
+import (
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/newrelic/infra-integrations-sdk/data/metric"
+	"github.com/newrelic/infra-integrations-sdk/integration"
+	"github.com/newrelic/nri-kubernetes/v2/src/definition"
+)
+
+// Asserter is a helper for checking whether an integration contains all the metrics defined in a specGroup.
+// It provides a chainable API.
+// Asserter is safe to use concurrently.
+type Asserter struct {
+	*sync.Mutex
+	entities        []*integration.Entity
+	specGroups      definition.SpecGroups
+	exclude         map[string]map[string]bool
+	excludeOptional bool
+	silent          bool
+}
+
+func NewAsserter() Asserter {
+	return Asserter{
+		Mutex: &sync.Mutex{},
+	}
+}
+
+// Using returns an asserter that will use the supplied specGroups to assert entities.
+func (a Asserter) Using(groups definition.SpecGroups) Asserter {
+	a.specGroups = groups
+	return a
+}
+
+// On returns an asserter configured to check for existence on the supplied entities.
+func (a Asserter) On(entities []*integration.Entity) Asserter {
```

Comment:
same here, without entities the tests is basically no-op

Reply:
You are right, it makes sense for the asserter to fail if there are either no entities or no groups. I will patch that!